### ISSUE TITLE
plugin not found in $PATH

### ISSUE THREAD

**Issue by iamnoah** [bufbuild/buf#1850]
buf 1.14.0 is not finding plugins added to the PATH variable:

```
> PATH="$(pwd)/tools:${PATH}" buf generate
Failure: plugin go-grpc: exec: "": executable file not found in $PATH

> ls tools
buf			protoc-gen-go		protoc-gen-go-grpc
```

We use this approach to pin plugin versions. Downgrading to to 1.13.1 is our workaround for now.

**Comment by bufdev** [comment]
Can you provide some more information for us to reproduce?

- Confirm that protoc-gen-go-grpc is executable
- OS, Arch, go env

**Comment by bufdev** [comment]
I've tried extensively and I am unable to reproduce this. I have to close for now, but please re-open if you can provide a specific, reproducible example, including os/arch information, the buf.gen.yaml you are using, and a test `.proto` file that fails.

**Comment by ohouse-andy** [comment]
Hi,

I have similar problem, downgrade to buf v1.13.1 helped, v1.14 and 1.15 do not work...

OS = Mac OS 13.1(22C65) Ventura
arch =  arm64

protoc-gen-go-grpc works well:
```
$ protoc-gen-go-grpc --version
protoc-gen-go-grpc 1.3.0
```

go env:
```
GO111MODULE=""
GOARCH="arm64"
GOBIN=""
GOCACHE="/Users/user/Library/Caches/go-build"
GOENV="/Users/user/Library/Application Support/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="arm64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/user/go/pkg/mod"
GONOPROXY="github.com/myorgname"
GONOSUMDB="go.myorgname.com"
GOOS="darwin"
GOPATH="/Users/user/go"
GOPRIVATE="github.com/myorgname"
GOPROXY="https://go.myorgname.com,https://proxy.golang.org,direct"
GOROOT="/Users/user/sdk/go1.18.3"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/Users/user/sdk/go1.18.3/pkg/tool/darwin_arm64"
GOVCS=""
GOVERSION="go1.18.3"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/dev/null"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/h7/7hcd2nns3qjf_k3blr_65fvc0wy1jm/T/go-build956519854=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

buf.gen.yaml
```yaml
version: v1beta1
plugins:
  - name: go
    out: gen/go
    opt: paths=source_relative
  - name: go-grpc
    out: gen/go
    opt: paths=source_relative
  - name: grpc-gateway
    out: gen/go
    opt:
      - paths=source_relative
      - generate_unbound_methods=true
  - name: openapiv2
    out: docs/openapiv2
```

buf.yaml
```yaml
version: v1beta1
name: buf.build/example/test
deps:
  - buf.build/googleapis/googleapis
  - buf.build/grpc-ecosystem/grpc-gateway
build:
  roots:
    - .
lint:
  use:
    - MINIMAL
    - STYLE_BASIC
  except:
    - PACKAGE_DIRECTORY_MATCH
breaking:
  use:
    - FILE
```

test.proto

```proto
syntax = "proto3";

package text.v1;
option go_package="github.com/myorg/test/v1";

import "google/api/annotations.proto";
import "google/protobuf/struct.proto";
import "protoc-gen-openapiv2/options/annotations.proto";

message Request {
  string data = 1;
}

message Response {
  string data = 1;
}

service TestService {
  rpc Test (Request) returns (Response) {
    option (google.api.http) = {
      get: "/v1/test"
    };

    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
      parameters: {
        headers: {
          name: "header";
          description: "header";
          type: STRING,
        };
      };
    };
  }
}
```

(protoc-gen-go-grpc       protoc-gen-grpc-gateway  protoc-gen-openapiv2 are all installed and can be executed)

**Comment by johnpena** [comment]
I've also experienced this under very similar conditions mentioned by @ohouse-andy. buf v1.15.0, MacOS Ventura 13.1, arm64 arch, confirmed `protoc-gen-go-grpc` and `protoc-gen-grpc-gateway` are on my $PATH and are callable. Downgrading to 1.13.0 resolved the issue.

**Comment by bufdev** [comment]
This is fixed in #1893, we will cut a v1.15.1 release with this patch. Apologies for the inconvenience.

**Comment by bufdev** [comment]
This is fixed on v1.15.1.
